test/integration/apiserver/flowcontrol/concurrency_test.go

```diff
@@ -0,0 +1,249 @@
+	timeout                                         = time.Second * 10
+)
+
+func setup(t testing.TB) (*httptest.Server, *rest.Config, framework.CloseFunc) {
+	opts := framework.MasterConfigOptions{EtcdOptions: framework.DefaultEtcdOptions()}
+	opts.EtcdOptions.DefaultStorageMediaType = "application/vnd.kubernetes.protobuf"
+	masterConfig := framework.NewIntegrationTestMasterConfigWithOptions(&opts)
+	resourceConfig := master.DefaultAPIResourceConfigSource()
+	resourceConfig.EnableVersions(schema.GroupVersion{
+		Group:   "flowcontrol.apiserver.k8s.io",
+		Version: "v1alpha1",
+	})
+	masterConfig.GenericConfig.MaxRequestsInFlight = 5
+	masterConfig.GenericConfig.MaxMutatingRequestsInFlight = 5
+	masterConfig.GenericConfig.OpenAPIConfig = framework.DefaultOpenAPIConfig()
+	masterConfig.ExtraConfig.APIResourceConfigSource = resourceConfig
+	_, s, closeFn := framework.RunAMaster(masterConfig)
+
+	return s, masterConfig.GenericConfig.LoopbackClientConfig, closeFn
+}
+
+func TestPriorityLevelIsolation(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.APIPriorityAndFairness, true)()
+	// NOTE: disabling the feature should fail the test
+	_, loopbackConfig, closeFn := setup(t)
+	defer closeFn()
+
+	loopbackClient := clientset.NewForConfigOrDie(loopbackConfig)
+	noxu1Client := getClientFor(loopbackConfig, "noxu1")
+	noxu2Client := getClientFor(loopbackConfig, "noxu2")
+
+	priorityLevelNoxu1, _, err := createPriorityLevelAndBindingFlowSchemaForUser(loopbackClient, "noxu1")
+	require.NoError(t, err)
+	priorityLevelNoxu2, _, err := createPriorityLevelAndBindingFlowSchemaForUser(loopbackClient, "noxu2")
+	require.NoError(t, err)
+
+	wg := &sync.WaitGroup{}
+	// "elephant"
+	streamRequests(wg, 10, 100, func() {
+		_, err := noxu1Client.CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{})
+		require.NoError(t, err)
+	})
+
+	streamRequests(nil, 1, 100, func() {
+		_, err := noxu2Client.CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{})
+		require.NoError(t, err)
+	})
+
+	wg.Wait()
+
+	dispatchedCountNoxu1, err := getRequestCountOfPriorityLevel(loopbackClient, priorityLevelNoxu1.Name)
+	require.NoError(t, err)
+	dispatchedCountNoxu2, err := getRequestCountOfPriorityLevel(loopbackClient, priorityLevelNoxu2.Name)
+	require.NoError(t, err)
+
+	assert.Equal(t, 1000, dispatchedCountNoxu1)
+	assert.Equal(t, 100, dispatchedCountNoxu2)
+}
+
+func getClientFor(loopbackConfig *rest.Config, username string) clientset.Interface {
+	config := &rest.Config{
+		Host:        loopbackConfig.Host,
+		QPS:         -1,
+		BearerToken: loopbackConfig.BearerToken,
+		Impersonate: rest.ImpersonationConfig{
+			UserName: username,
+		},
+	}
+	return clientset.NewForConfigOrDie(config)
+}
+
+func getRequestCountOfPriorityLevel(c clientset.Interface, priorityLevelName string) (int, error) {
+	resp, err := c.CoreV1().
+		RESTClient().
+		Get().
+		RequestURI("/metrics").
```

Comment:
Could consider returning a `map[priority level name]int` instead of re-getting and searching each time?

Reply:
updated